CommunicationLayer/CommDeviceControl/UdpMessageForwarder.cpp

```diff
@@ -26,27 +26,39 @@
 #include <QByteArray>
 #include <QDebug>
 #include <QHostAddress>
+#include <QTextCodec>
+
 
 #include "InfrastructureLayer/Settings/I_Settings.h"
 #include "UdpMessageForwarder.h"
 
+namespace
+{
+    const QString EXCHANGE_NAME = QString("amq.direct");
+}
+
 UdpMessageForwarder::UdpMessageForwarder(I_Settings& settings)
-: ipAddress_(settings.ipAddress())
-, port_(settings.udpPort())
 {
-    socket_.bind(ipAddress_, port_);
+  QString queueName = settings.queueName();
+
+  routingKey = settings.routingKey();
+
+  channel = Channel::Create(settings.ipAddress().toStdString(), (int)settings.udpPort());
+
+  channel->DeclareQueue(queueName.toStdString(), false, true, false, false);
+
+  channel->BindQueue(queueName.toStdString(), EXCHANGE_NAME.toStdString(), routingKey.toStdString());
+
 }
 
 UdpMessageForwarder::~UdpMessageForwarder()
 {
+  // TODO What to do for destructor? Nothing?
```

Comment:
I think that this could maybe use some discussion (which is why I left it here so I wouldn't forget), I don't think that there needs to be any cleanup as far as I can see, but it is also an option to have it cleanup the queue. I think leaving the queue alone makes more sense but I also don't have the same broad scope of the project that you do.